src/IR/IR.hs

```diff
@@ -1,4 +1,5 @@
 {-# LANGUAGE DeriveDataTypeable #-}
+{-# LANGUAGE FlexibleInstances #-}
```

Comment:
Which definition requires this?

Reply:
`instance (Pretty Expr ()) where` (line 370) requires this.
Without it, I get the error:
```
• Illegal instance declaration for ‘Pretty (Expr ())’
    (All instance types must be of the form (T a1 ... an)
     where a1 ... an are *distinct type variables*,
     and each type variable appears at most once in the instance head.
     Use FlexibleInstances if you want to disable this.)
• In the instance declaration for ‘Pretty (Expr ())’typecheck
```
It doesn't like that I'm only implementing the Pretty typeclass for `Expr t `when `Expr t`'s type variable is unit.

Reply:
Ah right, got it!